src/test/java/api/loans/CheckOutCalculateDueDateTests.java

```diff
@@ -950,145 +759,10 @@ private void checkFixedDayOrTime(String servicePointId, String policyProfileName
-          } else {
-            OpeningDay nextOpeningDay = nextDayPeriod.getOpeningDay();
-            String nextDate = nextOpeningDay.getDate();
-            LocalDate localDate = LocalDate.parse(nextDate, DATE_TIME_FORMATTER);
-
-            if (nextOpeningDay.getAllDay()) {
-              return new DateTime(localDate.atTime(LocalTime.MIN).toString()).withZoneRetainFields(DateTimeZone.UTC);
-            } else {
-              OpeningHour openingHour = nextOpeningDay.getOpeningHour().get(0);
-              LocalTime startTime = LocalTime.parse(openingHour.getStartTime());
-              return new DateTime(LocalDateTime.of(localDate, startTime).toString()).withZoneRetainFields(DateTimeZone.UTC);
-            }
-          }
-        } else {
-          LocalTime offsetTime = LocalTime.now(ZoneOffset.UTC).plusMinutes(duration);
-          if (isInPeriodOpeningDay(currentOpeningDay.getOpeningHour(), offsetTime)) {
-            LocalDate localDate = LocalDate.parse(currentDate, DATE_TIME_FORMATTER);
-            return new DateTime(LocalDateTime.of(localDate, offsetTime).toString()).withZoneRetainFields(DateTimeZone.UTC);
-          } else {
-            OpeningDay nextOpeningDay = nextDayPeriod.getOpeningDay();
-            String nextDate = nextOpeningDay.getDate();
-            LocalDate localDate = LocalDate.parse(nextDate, DATE_TIME_FORMATTER);
-
-            if (nextOpeningDay.getAllDay()) {
-              return new DateTime(localDate.atTime(LocalTime.MIN).toString()).withZoneRetainFields(DateTimeZone.UTC);
-            } else {
-              OpeningHour openingHour = nextOpeningDay.getOpeningHour().get(0);
-              LocalTime startTime = LocalTime.parse(openingHour.getStartTime());
-              return new DateTime(LocalDateTime.of(localDate, startTime).toString()).withZoneRetainFields(DateTimeZone.UTC);
-            }
-          }
-        }
-      } else {
-        OpeningDay nextOpeningDay = nextDayPeriod.getOpeningDay();
-        String nextDate = nextOpeningDay.getDate();
-        LocalDate nextLocalDate = LocalDate.parse(nextDate, DATE_TIME_FORMATTER);
-
-        if (nextOpeningDay.getAllDay()) {
-          return new DateTime(nextLocalDate.atTime(LocalTime.MIN).toString()).withZoneRetainFields(DateTimeZone.UTC);
-        }
-        OpeningHour openingHour = nextOpeningDay.getOpeningHour().get(0);
-        LocalTime startTime = LocalTime.parse(openingHour.getStartTime());
-        return new DateTime(LocalDateTime.of(nextLocalDate, startTime).toString()).withZoneRetainFields(DateTimeZone.UTC);
-      }
-    }
-  }
-
-  /**
-   * Minor threshold when comparing minutes or milliseconds of dateTime
-   */
-  private DateTime getThresholdDateTime(DateTime dateTime) {
-    return dateTime
-      .withSecondOfMinute(0)
-      .withMillisOfSecond(0);
+    assertThat(ERROR_MESSAGE_DUE_DATE + expectedDueDate + ", actual due date is " + actualDueDate,
+      actualDueDate.compareTo(expectedDueDate) == 0);
```

Comment:
Can we use` equals()` here? I think it is clearer than `compareTo() == 0`.

Reply:
Replaced by equivalence check `isEqual`